framework/src/executioners/Transient.C

```diff
@@ -442,8 +466,42 @@ Transient::solveStep(Real input_dt)
   // Update warehouse active objects
   _problem.updateActiveObjects();
 
+  if (_prev_time == _time && _relax_factor != 1.0) // i.e. _pic_it > 0
```

Comment:
So, why did not you use `_pic_it > 0` here? I am going off of your comment. This way you have to have that weird `_prev_time(_time - 1.0)` initialization in the c-tor list which I do not like.

Reply:
`_prev_time` is needed for relaxing sub-app variables. Only the master-app keeps track of `_picard_it`--for the sub-app `_picard_it` always equals 0. (Unless the sub-app has a sub-app of its own).

Reply:
Ok. Then, remove this comment and replace is with a better explanation. It will only confuse people in future (there is big Executioner refactoring planned). And also add a comment where you initialize `_prev_time`, so there is an explanation of the magic constant.

NOTE: To me, it looks like we need `prePicardIteration` and `postPicardIteraction` where this and the other part would be placed. But I guess we will get those after the refactoring is done.